packages/std/src/binary.rs

```diff
@@ -505,51 +503,17 @@ mod tests {
 
     #[test]
     fn binary_implements_as_ref() {
-        // Can use as_ref (this we already get via the Deref implementation)
-        let data = Binary(vec![7u8, 35, 49, 101, 0, 255]);
-        assert_eq!(data.as_ref(), &[7u8, 35, 49, 101, 0, 255]);
-
-        let data = Binary(vec![7u8, 35, 49, 101, 0, 255]);
-        let data_ref = &data;
-        assert_eq!(data_ref.as_ref(), &[7u8, 35, 49, 101, 0, 255]);
-
-        // Implements as ref
-
-        // This is a dummy function to mimic the signature of
-        // https://docs.rs/sha2/0.10.6/sha2/trait.Digest.html#tymethod.digest
-        fn hash(data: impl AsRef<[u8]>) -> u64 {
-            let mut hasher = DefaultHasher::new();
-            data.as_ref().hash(&mut hasher);
-            hasher.finish()
-        }
-
-        let data = Binary(vec![7u8, 35, 49, 101, 0, 255]);
-        hash(data);
-
-        let data = Binary(vec![7u8, 35, 49, 101, 0, 255]);
-        let data_ref = &data;
-        hash(data_ref);
+        let want = &[7u8, 35, 49, 101, 0, 255];
+        let data = Binary(want.to_vec());
+        assert_eq!(want, AsRef::<[u8]>::as_ref(&data));
+        assert_eq!(want, AsRef::<[u8]>::as_ref(&&data));
```

Comment:
That diff is cool, thank you. `hash` was really just written to have an input for the trait.